github/migrations_source_import.go

```diff
@@ -0,0 +1,313 @@
+	//         import progress response will include push_percent, which is
+	//         the percent value reported by git push when it is "Writing
+	//         objects".
+	//     complete - the import is complete, and the repository is ready
+	//         on GitHub.
+	//
+	// If there are problems, you will see one of these in the status field:
+	//
+	//     auth_failed - the import requires authentication in order to
+	//         connect to the original repository. Make an "Update Existing
+	//         Import" request, and include vcs_username and vcs_password.
+	//     error - the import encountered an error. The import progress
+	//         response will include the failed_step and an error message.
+	//         Contact GitHub support for more information.
+	//     detection_needs_auth - the importer requires authentication for
+	//         the originating repository to continue detection. Make an
+	//         "Update Existing Import" request, and include vcs_username
+	//         and vcs_password.
+	//     detection_found_nothing - the importer didn't recognize any
+	//         source control at the URL.
+	//     detection_found_multiple - the importer found several projects
+	//         or repositories at the provided URL. When this is the case,
+	//         the Import Progress response will also include a
+	//         project_choices field with the possible project choices as
+	//         values. Make an "Update Existing Import" request, and
+	//         include vcs and (if applicable) tfvc_project.
+	Status        *string  `json:"status,omitempty"`
+	CommitCount   *int     `json:"commit_count,omitempty"`
+	StatusText    *string  `json:"status_text,omitempty"`
+	AuthorsCount  *int     `json:"authors_count,omitempty"`
+	ImportPercent *float64 `json:"import_percent,omitempty"`
+	URL           *string  `json:"url,omitempty"`
+	HTMLURL       *string  `json:"html_url,omitempty"`
+	AuthorsURL    *string  `json:"authors_url,omitempty"`
+	RepositoryURL *string  `json:"repository_url,omitempty"`
+	Message       *string  `json:"message,omitempty"`
+	FailedStep    *string  `json:"failed_step,omitempty"`
+}
+
+func (i Import) String() string {
+	return Stringify(i)
+}
+
+// SourceImportAuthor identifies an author imported from a source repository.
+type SourceImportAuthor struct {
+	ID         *int    `json:"id,omitempty"`
+	RemoteID   *string `json:"remote_id,omitempty"`
+	RemoteName *string `json:"remote_name,omitempty"`
+	Email      *string `json:"email,omitempty"`
+	Name       *string `json:"name,omitempty"`
+	URL        *string `json:"url,omitempty"`
+	ImportURL  *string `json:"import_url,omitempty"`
+}
+
+func (a SourceImportAuthor) String() string {
+	return Stringify(a)
+}
+
+// LargeFile identifies a file larger than 100MB found during a repository import.
+type LargeFile struct {
+	RefName *string `json:"ref_name,omitempty"`
+	Path    *string `json:"path,omitempty"`
+	OID     *string `json:"oid,omitempty"`
```

Comment:
Is it possible that `OID` is an `*int`?
Maybe a URL do the GitHub docs here too would be good in the comment.